File: src/main/java/com/wesker926/leetcode/algorithms/p0748/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0748.s1;

import java.util.Arrays;

/**
 * @author wesker.gh
 * @date 2021/12/10
 * @description 字母表 + 遍历
 */
public class Solution {
    public String shortestCompletingWord(String licensePlate, String[] words) {
        int[] o = new int[26];
        for (char c : licensePlate.toLowerCase().toCharArray()) {
            if (Character.isLetter(c)) {
                o[c - 'a']++;
            }
        }

        int[] t = new int[26];
        int idx = 0, min = 16;
        loop1:
        for (int i = 0; i < words.length; i++) {
            Arrays.fill(t, 0);
            for (char c : words[i].toCharArray()) {
                t[c - 'a']++;
            }

            for (int j = 0; j < 26; j++) {
                if (t[j] < o[j]) {
                    continue loop1;
                }
            }

            if (words[i].length() < min) {
                idx = i;
                min = words[i].length();
            }
        }

        return words[idx];
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0748/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/10
// @description: 字母表 + 遍历
package s1

import (
	"strings"
	"unicode"
)

func shortestCompletingWord(licensePlate string, words []string) string {
	idx, min := 0, 16
	o := make([]int, 26)
	for _, c := range strings.ToLower(licensePlate) {
		if unicode.IsLetter(c) {
			o[c-'a']++
		}
	}

loop:
	for i, word := range words {
		t := make([]int, 26)
		for _, c := range word {
			t[c-'a']++
		}

		for j := 0; j < 26; j++ {
			if t[j] < o[j] {
				continue loop
			}
		}

		if len(word) < min {
			idx = i
			min = len(word)
		}
	}

	return words[idx]
}
